src/main/java/hudson/remoting/Channel.java

```diff
@@ -829,8 +830,16 @@ public void terminate(IOException e) {
             } // JENKINS-14909: leave synch block
         } finally {
             if (e instanceof OrderlyShutdown) e = null;
-            for (Listener l : listeners.toArray(new Listener[0]))
-                l.onClosed(this, e);
+            for (Listener l : listeners.toArray(new Listener[0])) {
+                try {
+                    l.onClosed(this, e);
+                } catch (Throwable t) {
+                    LogRecord lr = new LogRecord(Level.SEVERE, "Listener {0} propagated an exception for channel {1}'s close: {2}");
+                    lr.setThrown(t);
+                    lr.setParameters(new Object[]{l, this, t.getMessage()});
```

Comment:
if you drop the stack trace in a JUL formatter you still get the message (I'm scarred by this).
If you drop logging the Throwable all together then I don't care about supporting those people and they need to learn.
JUL should log this atomically as the StreamHandler users a Writer that has internal locking semantics.  Again if this doesn't happen then IMHO its a buggy 3rd party logger implementation (likely not provided by the JDK).

Anyway - I'm not passionate enough about this here to vote -1 based on this